app/services/organizations/importers/google_csv_import_service.rb

```diff
@@ -3,41 +3,61 @@
 module Organizations
   module Importers
     class GoogleCsvImportService
+      Status = Data.define(:success?, :count, :no_match, :errors)
       def initialize(file)
         @file = file
         @organization = Current.organization
+        @count = 0
+        @no_match = []
+        @errors = []
       end
 
       def call
+        row_num = 1
         CSV.foreach(@file.to_path, headers: true, skip_blanks: true) do |row|
+          row_num += 1
           # Using Google Form headers
           email = row["Email"].downcase
           csv_timestamp = Time.parse(row["Timestamp"])
 
           person = Person.find_by(email:, organization: @organization)
           previous = FormSubmission.where(person:, csv_timestamp:)
-          next unless person && previous.empty?
 
-          latest_form_submission = person.latest_form_submission
-
-          if latest_form_submission.form_answers.empty?
-            create_form_answers(latest_form_submission, row)
+          if person.nil?
+            @no_match << [row_num, email]
+          elsif previous.present?
+            next
           else
-            create_form_answers(FormSubmission.create!(person:, csv_timestamp:), row)
+            latest_form_submission = person.latest_form_submission
+            ActiveRecord::Base.transaction do
+              # This checks for the empty form submission that is added when a person is created
+              if latest_form_submission.csv_timestamp.nil? && latest_form_submission.form_answers.empty?
+                latest_form_submission.update!(csv_timestamp:)
+                create_form_answers(latest_form_submission, row)
+              else
```

Comment:
can we add a comment on this `else` condition...something like if we get here it means that the person has a new entry in the CSV so we want to import the data...or the likes. I already forgot what getting here means and had to think for a second 👯‍♂️